models/demos/ttnn_falcon7b/tt/falcon_decoder.py

```diff
@@ -103,7 +103,6 @@ def __call__(
             residual,
             memory_config=self.model_config["DROPOUT_ADD_OUTPUT_MEMCFG"],
         )
-        ttnn.deallocate(residual)
```

Comment:
what was this change?

Reply:
The residual tensor here points to the input activation tensor. When we trace a model, we preallocate input activation buffers for it, which get reused for every iteration. Therefore we cannot deallocate the input activation buffer after just running the fwd pass once, since we need the buffer for the next fwd pass.

Reply:
I thought the output of the decode would write back into the input-activation tensor? We'd be forcing model-writers to re-write model implementation for the memory allocator if they wanted to opt into the trace feature?